src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -1374,7 +1374,9 @@ private async Task DowngradeLockAsync(Awaiter awaiter, bool upgradedStickyWrite,
             {
                 await beginAfterPrerequisite.ConfigureAwait(SynchronizationContext.Current is NonConcurrentSynchronizationContext);
             }
+#pragma warning disable CA1031 // Do not catch general exception types
             catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
```

Comment:
📝 This is a bit ugly. If you prefer, I'm happy to rebase this change to disable CA1031 rather than use all these suppressions.

Reply:
Yes, I suppose I'd prefer that. I'm pretty comfortable that contributions to this library will be responsible in how it catches the base `Exception` type.

Reply:
Done!